OCPQE-19123: Add LVMS Operator CI test jobs

Phaow opened the pull request:
Add LVMS CI tests

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-compact-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse abort

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-ipi-ovn-lvms-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f14

radeore commented on ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/ipi/ovn/lvms/cucushift-installer-rehearse-baremetalds-ipi-ovn-lvms-workflow.yaml:
```diff
@@ -0,0 +1,17 @@
+        IP_STACK=v4
+        NETWORK_TYPE=OVNKubernetes
+        VM_EXTRADISKS=true
+        VM_EXTRADISKS_LIST="vda vdb"
```
Just trying to understand, here we are not specifically setting `LVM_CLUSTER_DEVICE_PATH`?

Phaow replied:
Yeah, correct.

radeore commented on ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sno/lvms/cucushift-installer-rehearse-aws-ipi-sno-lvms-workflow.yaml:
```diff
@@ -0,0 +1,16 @@
+  as: cucushift-installer-rehearse-aws-ipi-sno-lvms
+  steps: 
+    pre:
+    - chain: cucushift-installer-rehearse-aws-ipi-sno-provision
```
General question, for LVMS on SNO we need  vm_instance with 8 vCpu, I usually overwrite vm_type: 'm6i.2xlarge' to setup OCP AWS flexy cluster, do we have this vm_type setup in this chain?

Phaow replied:
Good point, currently in this chain we don't overwrite vm_type, I'll overwrite it in our workflow.

Phaow commented on ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sno/lvms/cucushift-installer-rehearse-aws-ipi-sno-lvms-workflow.yaml:
```diff
@@ -0,0 +1,16 @@
+    post:
+    - chain: cucushift-installer-rehearse-aws-ipi-deprovision
+    env:
+      LVM_CLUSTER_DEVICE_PATH: "/dev/nvme2n1"
```
Hmm, for this workflow the SNO cluster have 4 disks, 2 is created when install `nvme0n1` and `nvme1n1`, we create 2 extra disks used for lvm test `nvme2n1` and `nvme3n1`.

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-ipi-ovn-lvms-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-aws-ipi-sno-lvms-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-aws-ipi-sno-lvms-f14

Phaow commented:
/retest

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-aws-ipi-sno-lvms-f14

Phaow commented:
/retest

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-aws-ipi-sno-lvms-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-ipi-ovn-lvms-f14

Phaow commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-ipi-ovn-lvms-f14

radeore commented:
/lgtm

Phaow commented:
@yunjiang29 @jianlinliu @liangxia @kuiwang02 Could you please help take a look when you get a chance? Thanks!

Phaow commented on ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-nightly.yaml:
```diff
@@ -181,6 +181,24 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-aws-ipi-registry-s3-cloudfront
+- as: aws-ipi-sno-lvms-f14
```
@jianlinliu Good point, yeah the LVMS Operator support multi arch, but currently it released as standlone and many cases will reconfig the storage maybe impact other subteam cases, so we add the jobs only execute LVMS cases and these 2 jobs execute both non-destructive and destructive test cases(in 30min). We use aws cover the arm64 and BM cover the amd64, it'll produce less costs than separate jobs `non-destructive` and `destructive`. WDYT?

jianlinliu replied:
okay, got it.

Phaow commented:
Rebase latest codes to sovle the merge conflicts.

Phaow commented:
/retest

jianlinliu commented:
/lgtm

liangxia commented on ci-operator/step-registry/storage/create/aws-extra-disks/storage-create-aws-extra-disks-commands.sh:
```diff
@@ -0,0 +1,133 @@
+# Create an AWS EBS volume, volInfraTag will help delete volume when deleting cluster 
+function create_ebs_volume(){
+  if [ $# -ne 6 ] ;then
+    echo "Usage: create_ebs_volume <volName> <availabilityZone> <volSize> <volType> <volInfraTag>" && return 1
```
`$#` check for 6 params, while the example show 5.  Is it expected ?

Phaow replied:
Good catch, yeah it should be 6 params, missed the `region`! Fixed, thanks!

liangxia commented:
/lgtm

Phaow commented:
/retest

Phaow commented:
Hello @jsafrane, I added some refs used for LVMS Operator tests, since the refs maybe only used by storage related components I added them in our folder `ci-operator/step-registry/storage`(The rehearse results all looks good). Could you please help take a look when you get a chance for the final lgtm/approve. Thanks!

Phaow commented:
/assign @jsafrane

Phaow commented:
/pj-rehearse ack

yunjiang29 commented:
/lgtm

jianlinliu commented:
/lgtm

jsafrane commented:
/approve
step-registry/storage changes
